### PR TITLE
Refactored JobState to have shorter names

### PR DESCRIPTION
**Pull request by dfitchett**
## What was the problem?

JobStates were very long.

## How does this fix it?[^1]

JobState names will be shorter, resulting in shorter metric names, log statements, and db entries. 

## Stacked on PR
- #2555

## How to test this PR
- pull code
- Unit Test:
  - from /domain-ee/ee-ep-merge-app run pytest
- Integration Tests:
  - run [base platform](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose#platform-base)
  - from /domain-ee/ee-ep-merge-app run pytest ./integration
- End to End tests:
  - run this [CI](https://github.com/department-of-veterans-affairs/abd-vro/actions/workflows/ee-ep-merge-end-to-end.yml) action to verify successful integration test
  - run end to end tests locally:
    - `export BIP_CLAIM_URL=mock-bip-claims-api:20300`
    - run bgs and bip services and mocks:
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew :app:dockerComposeUp`
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew -p mocks :dockerComposeUp`
    - `./gradlew :domain-ee:ee-ep-merge-app:endToEndTest`

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: domain-ee/ee-ep-merge-app/src/python_src/schema/merge_job.py [hunk 1]
```diff
@@ -10,22 +10,22 @@
...
-    RUNNING_MOVE_CONTENTIONS_FAILED_REMOVE_SPECIAL_ISSUE = 'RUNNING_MOVE_CONTENTIONS_FAILED_REMOVE_SPECIAL_ISSUE'
-    RUNNING_MOVE_CONTENTIONS_FAILED_REVERT_TEMP_STATION_OF_JURISDICTION = 'RUNNING_MOVE_CONTENTIONS_FAILED_REVERT_TEMP_STATION_OF_JURISDICTION'
-    RUNNING_CANCEL_CLAIM_FAILED_REVERT_TEMP_STATION_OF_JURISDICTION = 'RUNNING_CANCEL_CLAIM_FAILED_REVERT_TEMP_STATION_OF_JURISDICTION'
+    GET_PENDING_CLAIM_FAILED_REMOVE_SPECIAL_ISSUE = 'GET_PENDING_CLAIM_FAILED_REMOVE_SPECIAL_ISSUE'
+    GET_PENDING_CLAIM_CONTENTIONS_FAILED_REMOVE_SPECIAL_ISSUE = 'GET_PENDING_CLAIM_CONTENTIONS_FAILED_REMOVE_SPECIAL_ISSUE'
+    SET_TEMP_STATION_OF_JURISDICTION_FAILED_REMOVE_SPECIAL_ISSUE = 'SET_TEMP_STATION_OF_JURISDICTION_FAILED_REMOVE_SPECIAL_ISSUE'
+    MOVE_CONTENTIONS_FAILED_REMOVE_SPECIAL_ISSUE = 'MOVE_CONTENTIONS_FAILED_REMOVE_SPECIAL_ISSUE'
+    MOVE_CONTENTIONS_FAILED_REVERT_TEMP_STATION_OF_JURISDICTION = 'MOVE_CONTENTIONS_FAILED_REVERT_TEMP_STATION_OF_JURISDICTION'
+    CANCEL_CLAIM_FAILED_REVERT_TEMP_STATION_OF_JURISDICTION = 'CANCEL_CLAIM_FAILED_REVERT_TEMP_STATION_OF_JURISDICTION'
```
Comment (nanotone) [inline domain-ee/ee-ep-merge-app/src/python_src/schema/merge_job.py]: This is so Java-ish but, then again, DRYing it up won't actually reduce the LoC count.

**Review by nanotone — APPROVED** [review]
LGTM!

File: domain-ee/ee-ep-merge-app/src/python_src/schema/merge_job.py (hunk 1 above)
Comment (dfitchett) [reply domain-ee/ee-ep-merge-app/src/python_src/schema/merge_job.py]: Sorry can't help myself! 😆

**Review by raudabaugh — APPROVED** [review]

File: domain-ee/ee-ep-merge-app/src/python_src/util/custom_enum.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    StrEnum subclasses enum.Enum and overrides `enum.auto()` to have values equal to their names.
+    Enums inheriting from this class that set values using `enum.auto()` will have values equal to their names.
+    """
+
+    # noinspection PyMethodParameters
+    def _generate_next_value_(name, start, count, last_values) -> str:  # type: ignore
+        """
+        Uses the name as the automatic value, rather than an integer
+
+        See https://docs.python.org/3/library/enum.html#using-automatic-values for reference
```
Comment (nanotone) [inline domain-ee/ee-ep-merge-app/src/python_src/util/custom_enum.py]: Cute!

Yeah, this is a neat trick. This URL doesn't have the fragment for me though; https://docs.python.org/3/howto/enum.html#using-automatic-values is the one for me. 🤷

**Review by nanotone — APPROVED** [review]
👍

**Pull request merged**